src/api/firebase.js

```diff
@@ -205,3 +206,56 @@ export async function deleteItem() {
 	 * this function must accept!
 	 */
 }
+
+export async function getNextPurchaseEstimate(
+	listPath,
+	itemId,
+	currentDate,
+	purchaseTimestamp,
+) {
+	// reference the item path
+	const itemDocRef = doc(db, listPath, 'items', itemId);
+	try {
+		const itemDocSnap = await getDoc(itemDocRef);
+		if (itemDocSnap.exists()) {
+			const itemData = itemDocSnap.data();
+			// get previous purchase date or date item was created
+			const previousPurchase =
+				purchaseTimestamp !== null
+					? purchaseTimestamp.toDate()
+					: itemData.dateCreated.toDate();
+
+			console.log(
+				`Current item was previously purchased on ${previousPurchase.toLocaleString()}`,
+			);
+			// calculate the previous estimate and days since last purchase
+			const previousEstimate = getDaysBetweenDates(
+				previousPurchase,
+				itemData.dateNextPurchased.toDate(),
+			);
+			const daysSinceLastPurchase = getDaysBetweenDates(
+				previousPurchase,
+				currentDate,
+			);
+
+			// calculate the next purchase estimate
+			const nextPurchaseEstimatedDays = calculateEstimate(
+				previousEstimate,
+				daysSinceLastPurchase,
+				itemData.totalPurchases,
+			);
+
+			const nextPurchaseEstimate = getFutureDate(nextPurchaseEstimatedDays);
+			// update the document with the new next purchase date
+			await updateDoc(itemDocRef, {
+				dateNextPurchased: nextPurchaseEstimate,
+			});
+
+			return `Estimated amount of days until next purchase is ${nextPurchaseEstimatedDays} day(s)`;
+		} else {
+			throw new Error(`Item with ID ${itemId} not found in list ${listPath}`);
```

Comment:
Would it be better to tell the user the item name instead of itemId?

Reply:
That's great question! When this function errors out the user gets an error from ListItem component:
<code>alert(`Item was not marked as purchased`, error);</code>
I can add an item name to this alert!